### DISCUSSION TITLE
How to upgrade EJBCA to latest version

### DISCUSSION THREAD

**Discussion by huang714669** [Keyfactor/ejbca-ce#68] [General]
Currently, we use EJBCA with version 6.1.0.4, how to upgrade to the latest version base on the older version? 

Ideally, I hope the latest version is Compatible with my older versions (we can use the same ejbcaws client as before, and the database schema will have no change as well）.

**Comment by primetomas** [comment]
Yes, the WebService API should be backwards compatible, going decades back. The database have updated schema for sure, for the many new features. Your old database will be upgraded with the schema updates that are part of the upgrade steps though, so you're fine. Make a backup of your database before starting the upgrade, there are some steps going back all that time. We can upgrade all the way from EJBCA 4 though, althought the farther you go back the more chance of hickups of course.

The upgrade is documented here, with all the steps and possible intermediates you need to go through:
https://doc.primekey.com/ejbca/ejbca-installation/upgrading-ejbca
